proto/buf/alpha/registry/v1alpha1/search.proto

```diff
@@ -86,6 +96,10 @@ message SearchRequest {
   uint32 page_size = 2;
   // The first page is returned if this is 0.
   uint32 page_token = 3;
+  // Empty list means show all. Supplying one or more enums will 
+  // limit the search to the only the requested resources. 
+  // Supplying all possible enums is equivalent to empty list.
+  repeated SearchFilter filters = 4;
```

Comment:
So multiple enums translate into a logical `AND`? I don't understand how this would be meaningful for the `enum SearchFilter`, because no result can be User and Plugin at the same time.

It could be a logical `OR`, where supplying enum values returns only results that match one or another of the given resource types. Supplying all possible enums would show all results. 

However, I am not sure if we really need that filter capability in the search UI. Instead of going through multiple checkboxes for resource types, the user could just as well go through multiple radio options. For example, see the resource type selection in [GitHub's search](https://github.com/bufbuild/buf/search?q=foo) (sidebar on the left).

The repeated enum with `OR` behavior also adds ambiguity, for example `SEARCH_FILTER_UNSPECIFIED` in a request field, and inconvenient behavior of an empty filter list, so I think it might actually be best to go with a simple version for now, with an unrepeated field, and maybe rename `enum SearchFilter` to something more specific like `SearchResourceType`.

Reply:
multiple enums does translate to logical `OR` - when we say "Supplying all possible enums is equivalent to empty list." we mean "Supplying all possible enums is equivalent to empty list of enums." not "Supplying all possible enums is equivalent to empty list of results." Do you think that would be clearer? 

I think @SyrieBianco envisions the filter capability in a future search UI, not necessarily the current searchbar as it exists today. But I understand your point with respect to the Github search, that search implementation would actually be possible with the endpoints that exist today (`SearchPlugins`, `SearchTemplates`, etc)

Reply:
Ah, I misread that as _result_ list. @mfridman gave me some background on the filter enum that sounds very promising, and rebuts my nits.

Reply:
Updated the comment for clarity!!